### PR TITLE
Be resilient to certain missing fields in metadata.json

### PR DESCRIPTION
**Pull request by Wyverald**
Including `yanked_versions` and `maintainers`

Fixes https://github.com/bazel-contrib/bcr-ui/issues/134

### CODE REVIEW

File: data/utils.ts [hunk 1]
```diff
@@ -17,14 +17,14 @@ export const BUILDOZER_BIN = path.join(process.cwd(), 'bin', 'buildozer')
 
 export interface Metadata {
   homepage?: string
-  maintainers: Array<{
```
Comment (alexeagle) [inline data/utils.ts:20]: nit: we could consider doing the opposite. We already have a schema validator in the bcr that checks all module metadata is conformant.

(and this Metadata schema ought to be imported from that JSON schema so we know the two repos agree on required fields, etc)

WDYT?

Comment (Wyverald) [reply data/utils.ts:20]: sure -- I just have no idea how to do any of that :)

I suggest we submit this PR first to make sure we have an updated website build, and then worry about finding engineering time for your suggestion.

Comment (hobofan) [reply data/utils.ts:20]: I'm also more in favor of having a JSON schema validation in BCR, which can serve as a source of truth, as that also would make implementation of alternative registry UIs easier.

But yeah, let's go ahead and merge this for now, as this seems to be something that happens somewhat regularly now.

**Review by hobofan — APPROVED** [review]

File: data/utils.ts (hunk 1 above)
Comment (hobofan) [reply data/utils.ts:20]: Opened up an issue for it here: https://github.com/bazelbuild/bazel-central-registry/issues/1365

**Comment by hobofan** [comment]
I open up an issue to use the upstream schema as the basis for our types, to prevent format reading mistakes in the future: https://github.com/bazel-contrib/bcr-ui/issues/137

**Pull request merged**